lms/db/__init__.py

```diff
@@ -46,14 +46,13 @@ def update_from_dict(self, data, skip_keys=None):
         if skip_keys is None:
             skip_keys = {"id"}
 
-        columns = {col.key for col in self.iter_columns()}
-        if skip_keys:
-            if not isinstance(skip_keys, set):
-                raise TypeError(
-                    f"Expected a set of keys to skip but found '{type(skip_keys)}'"
-                )
+        if not isinstance(skip_keys, set):
+            raise TypeError(
+                f"Expected a set of keys to skip but found '{type(skip_keys)}'"
+            )
```

Comment:
It was possible before to squeeze through this condition with `skip_keys` being an empty dict. It seems better to me to just fail.

So this eliminates a code path, and probably a bug too.